python/tchannel/tornado/connection.py

```diff
@@ -222,6 +223,17 @@ def _loop(self):
                     future = self._outstanding.get(context.message_id)
                 else:
                     future = self._outstanding.pop(context.message_id)
+
+                    if context.message.message_type == Types.ERROR:
+                        protocol_error = (
+                            self.response_message_factory.build_protocol_error(
+                                context.message,
+                                context.message_id,
```

Comment:
Unrelated, what's to stop us from putting `message_id` on the `message`? That seems to be the primary use for `context` right now.